tfhe/src/shortint/server_key/compressed.rs

```diff
@@ -403,3 +405,76 @@ impl CompressedServerKey {
         }
     }
 }
+
+impl ParameterSetConformant for ShortintCompressedBootstrappingKey {
+    type ParameterSet = PBSConformanceParameters;
+
+    fn is_conformant(&self, parameter_set: &Self::ParameterSet) -> bool {
+        match (self, parameter_set.multi_bit) {
+            (Self::Classic(a), None) => {
+                let param: BootstrapKeyConformanceParams = parameter_set.into();
+
+                a.is_conformant(&param)
+            }
+            (
+                Self::MultiBit {
+                    seeded_bsk,
+                    deterministic_execution: _,
+                },
+                Some(_grouping_factor),
+            ) => {
+                let param: MultiBitBootstrapKeyConformanceParams =
+                    parameter_set.try_into().unwrap();
+
+                seeded_bsk.is_conformant(&param)
+            }
+            _ => false,
+        }
+    }
+}
+
+impl ParameterSetConformant for CompressedServerKey {
+    type ParameterSet = PBSParameters;
+
+    fn is_conformant(&self, parameter_set: &Self::ParameterSet) -> bool {
+        let Self {
+            key_switching_key,
+            bootstrapping_key,
+            message_modulus,
+            carry_modulus,
+            max_degree,
+            max_noise_level,
+            ciphertext_modulus,
+            pbs_order,
+        } = self;
+
+        let params: PBSConformanceParameters = parameter_set.into();
+
+        let pbs_key_ok = bootstrapping_key.is_conformant(&params);
+
+        let param: KeyswitchKeyConformanceParams = parameter_set.into();
+
+        let ks_key_ok = key_switching_key.is_conformant(&param);
+
+        let expected_max_degree = MaxDegree::integer_radix_server_key(
+            parameter_set.message_modulus(),
+            parameter_set.carry_modulus(),
+        );
```

Comment:
not true in the general case but ok for the blockchain case, mark as TODO here

Reply:
Done

Reply:
Moved to integer level by adding `expected_max_degree` to conformance parameter set